### PR TITLE
genpolicy: Update oci_distribution to 0.10.0

### PR DESCRIPTION
**Pull request by Redent0r**
* Bring back remaining changes from `cc-msft-prototypes` (Only https://github.com/microsoft/kata-containers/commit/da40e4150a00856aaadbd70d64b333cd9d249b38)

* Update oci_distribution and support alternative layer media type. These 2 things add back support for images like `docker.io/library/busybox:latest`

test run: 100% pass

### CODE REVIEW

**Review by ms-mahuber — APPROVED** [review]

**Review by danmihai1 — CHANGES_REQUESTED** [review]
Please port the remaining genpolicy commits from cc-msft-prototypes into msft-main, before making these oher changes - to make sure we don't leave useful changes behind, in cc-msft-prototypes.

File: src/tools/genpolicy/src/registry.rs [hunk 1]
```diff
@@ -232,7 +232,10 @@ async fn get_image_layers(
     for layer in &manifest.layers {
         if layer
             .media_type
-            .eq(manifest::IMAGE_DOCKER_LAYER_GZIP_MEDIA_TYPE)
```
Comment (danmihai1) [inline src/tools/genpolicy/src/registry.rs:235]: Is manifest::IMAGE_LAYER_GZIP_MEDIA_TYPE support a new feature of the latest oci_distribution?

- If it's an already existing feature, I think you should use a separate PR for this change.
- If it's new, then using a single commit seems reasonable to me.

**Comment by danmihai1** [comment]
> Update oci_distribution client and support alternative layer media type. This adds back support for images like `docker.io/library/busybox:latest`
> 
> test run: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=472233&view=results (100% pass)

Please don't include MSFT private links in public PRs.

**Review by danmihai1 — CHANGES_REQUESTED** [review]

**Comment by danmihai1** [comment]
I believe the crate name is oci_distribution, not oci_distribution_client

File: src/tools/genpolicy/src/registry.rs [hunk 2]
```diff
@@ -232,7 +232,10 @@ async fn get_image_layers(
...
+            .eq(manifest::IMAGE_DOCKER_LAYER_GZIP_MEDIA_TYPE) ||
```
Comment (danmihai1) [inline src/tools/genpolicy/src/registry.rs:235]: Please format this code using rust-analyzer. It looks like it wants to format it as:

        if layer
            .media_type
            .eq(manifest::IMAGE_DOCKER_LAYER_GZIP_MEDIA_TYPE)
            || layer.media_type.eq(manifest::IMAGE_LAYER_GZIP_MEDIA_TYPE)

**Review by danmihai1 — CHANGES_REQUESTED** [review]

**Comment by Redent0r** [comment]
> Please port the remaining genpolicy commits from cc-msft-prototypes into msft-main, before making these oher changes - to make sure we don't leave useful changes behind, in cc-msft-prototypes.

Ported only change missing from cc-msft-prototypes (https://github.com/microsoft/kata-containers/commit/da40e4150a00856aaadbd70d64b333cd9d249b38) and rebased my changes on top of it to preserve chronological order

File: src/tools/genpolicy/src/registry.rs (hunk 2 above)
Comment (Redent0r) [reply src/tools/genpolicy/src/registry.rs]: Thank you for highlighting this. I had rust-analyzer, but didn't have `formatOnSave` toggled on my vs code settings

**Review by danmihai1 — APPROVED** [review]

**Pull request merged**